packages/medusa/src/repositories/product.ts

```diff
@@ -26,193 +28,109 @@ export type FindWithoutRelationsOptions = DefaultWithoutRelations & {
   where: DefaultWithoutRelations["where"] & {
     price_list_id?: FindOperator<PriceList>
     sales_channel_id?: FindOperator<SalesChannel>
+    discount_condition_id?: string
   }
 }
 
 @EntityRepository(Product)
 export class ProductRepository extends Repository<Product> {
-  private mergeEntitiesWithRelations(
-    entitiesAndRelations: Array<Partial<Product>>
-  ): Product[] {
-    const entitiesAndRelationsById = groupBy(entitiesAndRelations, "id")
-    return map(entitiesAndRelationsById, (entityAndRelations) =>
-      merge({}, ...entityAndRelations)
-    )
-  }
-
-  private async queryProducts(
-    optionsWithoutRelations: FindWithoutRelationsOptions,
-    shouldCount = false
+  public async findWithRelationsAndCount(
```

Comment:
thought: This is quite a large refactor for adding the support to filter for discount condition id. Would it make sense to split this into several PRs to make it easier to digest for reviewers? I would assume we could add support for discount condition id similar to how we handle filtering for pricelist, but please correct me if I am wrong.

Reply:
you are right, I just used the repository utils instead of what was present like I ve done for the other pr's thinking that it was alright. But since the product repo is bigger it appears at a lot of changes 👍 I'll split it